buku

```diff
@@ -2129,16 +2129,21 @@ class BukuDb:
             LOGERR('delete_rec_all(): %s', e)
             return False
 
-    def cleardb(self):
+    def cleardb(self, confirm=True):
         """Drops the bookmark table if it exists.
 
+        Parameters
+        ----------
+        confirm : bool
+            False if confirmation prompt is required. Default is True.
+
         Returns
         -------
         bool
             True on success, False on failure.
         """
 
-        resp = read_in('Remove ALL bookmarks? (y/n): ')
+        resp = ('y' if not confirm else read_in('Remove ALL bookmarks? (y/n): '))
```

Comment:
This is a _much_ better way of using it in non-CLI mode than using hacky workarounds to temporarily disable terminal I/O :melting_face: